### PR TITLE
add deployments table

### PR DESCRIPTION
**Pull request by judell**
```
select
  name as project,
  state,
  url,
  creator ->> 'email' as creator,
  meta ->> 'githubCommitMessage' as commit_message,
  meta ->> 'githubCommitRef' as commit_ref
from
  vercel_deployment
order by 
  created_at desc
 ```
  
```
 +------------------+-------+-------------------------------------------+-------------------+--------------------------------------------------
| project          | state | url                                       | creator           | commit_message                                   
+------------------+-------+-------------------------------------------+-------------------+--------------------------------------------------
| hub-steampipe-io | READY | hub-steampipe-rk6tppql3-turbot.vercel.app | david@boeke.com   | Merge pull request #590 from turbotio/staging    
|                  |       |                                           |                   |                                                  
|                  |       |                                           |                   | graphics for gitguardian plugin, closes #586     
| hub-steampipe-io | READY | hub-steampipe-cmeeqlsyx-turbot.vercel.app | david@boeke.com   | Merge pull request #590 from turbotio/staging    
|                  |       |                                           |                   |                                                  
|                  |       |                                           |                   | graphics for gitguardian plugin, closes #586     
| hub-steampipe-io | READY | hub-steampipe-a2g4spfh6-turbot.vercel.app | david@boeke.com   | Merge pull request #590 from turbotio/staging    
|                  |       |                                           |                   |                                                  
|                  |       |                                           |                   | graphics for gitguardian plugin, closes #586     
| hub-steampipe-io | READY | hub-steampipe-r17f4ofwi-turbot.vercel.app | david@boeke.com   | Merge pull request #590 from turbotio/staging    
|                  |       |                                           |                   |                                                  
|                  |       |                                           |                   | graphics for gitguardian plugin, closes #586     
| steampipe-io     | READY | steampipe-h3ow7q9g7-turbot.vercel.app     | jon@jonudell.info | add cta (#594)                                   
```

### CODE REVIEW

**Comment by judell** [comment]
@ved, @e-gineer: We had been using the `latest_deployments` property of the `vercel_project` table to track deployments but the information there was incomplete. The Go SDK did not provide a `deployment` table, and it is now archived, so for now I've added it in a fork of Nathan's fork of the original, not sure whether/how to consolidate. (https://github.com/e-gineer/vercel-go/compare/main...judell:vercel-go:main)

I still need to add a doc for the new table, will do that now using the above example.

File: docs/tables/vercel_deployment.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  name as project,
+  state,
+  url,
+  creator ->> 'email' as creator,
+  meta ->> 'githubCommitMessage' as commit_message,
+  meta ->> 'githubCommitRef' as commit_ref
+from
+  vercel_deployment
+order by 
+  created_at desc
```
Comment (misraved) [inline docs/tables/vercel_deployment.md]:
```suggestion
  created_at desc;
```

**Comment by misraved** [comment]
@judell I believe the table supports pagination and we could add support for the same 👍 .

**Comment by judell** [comment]
Hey @misraved, the pagination pattern for the existing tables was like this:

```
		if res.Pagination.Next == 0 {
			break
		}
		opts.Until = int(res.Pagination.Next)
```

When I tried that for the new table, it just kept streaming until I stopped it. I think this wasn't noticed because the existing tables set the gulp size to 100, and none have that many records. But there are lots of deployments.

I'm looking into how to handle limiting -- from both the SQL and Go perspectives. Good thing for me to learn, then explain.

**Comment by judell** [comment]
The method implemented here -- https://github.com/turbot/steampipe-plugin-vercel/blob/add-deployments-table/vercel/table_vercel_deployment.go#L42 -- seems to work. I don't think I've seen this kind of handling of the Postgres LIMIT in other plugins so maybe I'm doing it wrong though?

There are at least 250K records in that table, I'm not done listing them as I write this, so that makes this also an opportunity to learn (and explain) about pushing quals down into the plugin. We should be able to use the API's `since` and `until` params to map SQL time constraints to API time constraints, right?

**Comment by judell** [comment]
Progress! Now `created_at` is a KeyColumn, and you can do queries like:

```
select * from vercel_deployment where created_at > '2022-10-14T1:41:18'::timestamptz limit 3   
````

```
select count(*) from vercel_deployment where created_at between '2022-10-01' and '2022-10-31'
```

This has been a great learning exercise for me. Until now I didn't have a clear picture of how we translate (for example) Postgres time constraints to API time constraints. Finally it's sinking in, and now I'll be able to write the docs for this that I've been wanting to write.

**Review by misraved — APPROVED** [review]

**Pull request merged**